packages/main/src/SegmentedButton.js

```diff
@@ -213,7 +218,25 @@ class SegmentedButton extends UI5Element {
 			this._selectItem(event);
 		} else if (isSpace(event)) {
 			event.preventDefault();
+		} else if (isPageUp(event)) {
+			event.preventDefault();
+			this._onPageUpOrDown(isPageUp(event));
+		} else if (isPageDown(event)) {
+			event.preventDefault();
+			this._onPageUpOrDown();
+		}
+	}
+
+	_onPageUpOrDown(pageUpPressed) {
```

Comment:
This function doesn't make the code more readable or structured. Better execute the code on place.

Reply:
Done